src/zarate.jl

```diff
@@ -105,34 +115,45 @@ function solve_positions(
 end
 
 """
-    ordering_problem(::Zarate, graph, layer2nodes; force_order)
+    ordering_problem(::Zarate, graph, layer2nodes; force_order, force_equal_layers)
 
 Formulates the problem of working out optimal ordering as a MILP.
 
 Returns:
  - `model::Model`: the JuMP model that when optized will find the optimal ordering
- - `is_before::AbstractVector{AbstractVector{Variable}}`: the variables of the model,
-   which once solved will have `value(is_before[n1][n2]) == true`
-   if `n1` is best arrange before `n2`.
+ - `is_before::Dict{Pair{Int, Int}, VariableRef}`: the variables of the model,
+   which once solved will have `value(is_before[n1=>n2]) == true` if `n1` is best
+   arranged before `n2`.
 """
 function ordering_problem(layout::Zarate, graph, layer2nodes;
-        force_order=Vector{Pair{Int, Int}}())
+        force_order=Vector{Pair{Int, Int}}(),
+        force_equal_layers=Vector{Pair{Int, Int}}())
     m = Model(layout.ordering_solver)
     set_silent(m)
 
-    node_is_before = Vector{Any}(undef, nv(graph))
+    is_before = Dict{Pair{Int, Int}, VariableRef}()
     for (layer, nodes) in enumerate(layer2nodes)
-        before = @variable(m, [nodes, nodes], Bin, base_name="befores_$layer")
+        # initialize `is_before` variable-container
+        for n1 in nodes
+            for n2 in nodes
+                n1 === n2 && continue
+                haskey(is_before, (n1, n2)) && continue
+                is_before[n1=>n2] = @variable(m, binary=true,
+                                                base_name="befores_$layer[$n1=>$n2]")
+            end
+        end
+        # constrain variables
         for n1 in nodes
-            node_is_before[n1] = before[n1, :]
             for n2 in nodes
                 n1 === n2 && continue
                 # can't have n1<n2 and n2<n1
-                @constraint(m, before[n1, n2] + before[n2, n1] == 1)
+                @constraint(m, is_before[n1=>n2] + is_before[n2=>n1] == 1)
                 for n3 in nodes
                     (n1 === n3 || n2 === n3) && continue
                     # at most two of these 3 hold
-                    @constraint(m , before[n1, n2] + before[n2, n3] + before[n3, n1] <= 2)
+                    @constraint(m, is_before[n1=>n2] +
+                                   is_before[n2=>n3] + 
+                                   is_before[n3=>n1] <= 2)
```

Comment:
There is no need to break this on to multiple lines
```suggestion
                    @constraint(m, is_before[n1=>n2] + is_before[n2=>n3] +  is_before[n3=>n1] <= 2)
```